test/component_test.rb

```diff
@@ -114,6 +116,40 @@ class ComponentTest < ActiveSupport::TestCase
     assert_equal [], component.foos
   end
 
+  test "initialize with given attribute and successfull validation" do
+    component_class = Class.new(Components::Component) do
+      attribute :foo
+      validates :foo, presence: true
+    end
+
+    assert_nothing_raised { component_class.new(:view, foo: "bar") }
+  end
+
+  test "initialize without attribute and failing validation" do
+    component_class = Class.new(Components::Component) do
+      attribute :foo
+      validates :foo, presence: true
+
+      # supplies a name argument for active validations on anonymous classes
+      def self.model_name
+        ActiveModel::Name.new(self, nil, "temp")
+      end
+    end
+
+    exception = assert_raises { component_class.new(:view) }
+    assert_instance_of ActiveModel::ValidationError, exception
```

Comment:
You can assert the class with `assert_raises` I think: `assert_raises ActiveModel::ValidationError { component_class.new(:view) }`.

Reply:
It takes a pair of brackets - but it works fine:

`assert_raises(ActiveModel::ValidationError) { component_class.new(:view)`